superdesk/publish/formatters/__init__.py

```diff
@@ -147,14 +147,15 @@ def get_formatter(format_type, article):
 
     :param etree: parsed xml
     """
-    for formatter in formatters:
-        if formatter.can_format(format_type, article):
-            return formatter
+    for formatter_cls in formatters:
+        formatter_instance = formatter_cls()
+        if formatter_instance.can_format(format_type, article):
```

Comment:
It would be possible to run `can_format`  without creating an `formatter_instance`, but since `can_format` is not static/class method and all core's + custom use it, it's better not to touch it.

Reply:
yep would keep it bc compatible, but we could introduce new static method which could be implemented in formatter, and default implementation could be creating instance and calling that can_format method

Reply:
@petrjasek we'll still create an instance to run `can_format`, so there is no actual benefit imo. It will make only `get_formatter` function just a little bit cleaner and at the same time that new static method can confuse somebody. I'm not sure :(